### PR TITLE
Improvements to main application with empahsis on documentation and usability

### PR DESCRIPTION
**Pull request by screamerbg**
Also:
* updated bootloaders and main application configuration in preparation for Mbed OS 5.11.
* new targets support

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -98,23 +99,58 @@ void button_callback(MbedCloudClientResource *resource, const NoticationDelivery
...
-    printf("Connecting to the network...\n");
-
-    // Connect to the internet (DHCP is expected to be on)
-    nsapi_error_t status = net->connect();
+    printf("\nStarting Simple Pelion Device Management Client example\n");
+
+#ifdef USE_BUTTON
+    // If the User button is pressed ons start, then format storage.
+    DigitalIn *user_button = new DigitalIn(BUTTON1);
+    const int PRESSED = 1;
```
Comment (MarceloSalazar) [inline main.cpp]: Looks to me BUTTON1 is already defined here https://github.com/ARMmbed/pelion-ready-example/pull/49/files#diff-118fcbaaba162ba17933c7893247df3aR42

Comment (screamerbg) [reply main.cpp]: As discussed, BUTTON1 is initialized as DigitalIn to read the current state of the button and as InterruptIn to trigger callback function

Comment (janjongboom) [reply main.cpp]: You can read from InterruptIn, no need for DigitalIn.

File: main.cpp [hunk 2]
```diff
@@ -98,23 +99,58 @@ void button_callback(MbedCloudClientResource *resource, const NoticationDelivery
...
+    if (user_button->read() == PRESSED) {
```
Comment (janjongboom) [inline main.cpp]: We already have format code in Simple Cloud Client. Also why do we always need to format storage, this should never be necessary. Why can't this be fixed upstream.

File: main.cpp [hunk 3]
```diff
@@ -124,35 +160,43 @@ int main(void) {
...
 
     // Callback that fires when registering is complete
     client.on_registered(&registered);
 
-    // Register with Pelion Device Management
+    // Register with Pelion DM
     client.register_and_connect();
 
-    // Placeholder for callback to update local resource when GET comes.
+    int i = 600; // wait up 60 seconds before attaching sensors and button events
```
Comment (janjongboom) [inline main.cpp]: Why is this even necessary?!

File: main.cpp [hunk 4]
```diff
@@ -124,35 +160,43 @@ int main(void) {
...
+    while (i-- > 0 && !client.is_client_registered()) {
+        wait_ms(100);
+    }
+
+#ifdef USE_BUTTON
```
Comment (janjongboom) [inline main.cpp]: BUTTON1 is always defined in the HAL these days, so this `USE_BUTTON` macro should not be needed.

File: main.cpp [hunk 5]
```diff
@@ -61,27 +70,19 @@ void pattern_updated(MbedCloudClientResource *resource, m2m::String newValue) {
  *               Note that the buffer is deallocated after leaving this function, so copy it if you need it longer.
  * @param size Size of the body
  */
-void blink_callback(MbedCloudClientResource *resource, const uint8_t *buffer, uint16_t size) {
```
Comment (janjongboom) [inline main.cpp]: I don't agree on removing this feature.

**Comment by janjongboom** [comment]
I'm not happy that this adds 50 lines of code (33% more!) on main.cpp.

**Comment by janjongboom** [comment]
@screamerbg What do you think about the above? Also needs https://github.com/ARMmbed/simple-mbed-cloud-client/pull/66

**Pull request merged**